src/roadmap/models.py

```diff
@@ -72,3 +90,25 @@ class TaggedParagraph(BaseModel):
     title: str = Field(description="The paragraph title")
     text: str = Field(description="The paragraph text")
     tag: str = Field(description="The paragraph htmltag")
+
+
+def _calculate_support_status(
+    start_date: date | str | None, end_date: date | str | None, current_date: date
```

Comment:
This is a really nice refactor. It even makes the code easier to test because you can supply the date during testing (dependency injection).